### PR TITLE
feat(core)!: make crossterm optional

### PR DESCRIPTION
**Pull request by mikevoronov**
Only for publishing

### CODE REVIEW

File: ccp/src/utility_thread/thread.rs [hunk 1]
```diff
@@ -101,24 +101,39 @@ impl UtilityThreadImpl {
...
+        #[cfg(feature = "crossterm")]
+        {
+            use futures::FutureExt;
+            use futures::StreamExt;
+
+            loop {
+                tokio::select! {
+                Some(message) = self.to_utility.recv() => self.handle_to_utility_message(message).await,
+                _ = cum_hashrate_ticker.tick() => self.handle_cum_hashrate_tick().await,
+                maybe_event = terminal_event_reader.next().fuse() => self.handle_terminal_event(maybe_event).await,
```
Comment (monoid) [inline ccp/src/utility_thread/thread.rs:123]: Duplicate code is hell to maintain.
I may suggest defining
```rust
        #[cfg(not(feature = "crossterm"))]
        let mut terminal_event_reader = futures::stream::pending();
```
and some dummy implementation of `handle_terminal_event`.

Comment (mikevoronov) [reply ccp/src/utility_thread/thread.rs:123]: ofc, this PR is just for testing purposes

**Review by monoid — APPROVED** [review]
utACK

File: ccp/src/utility_thread/thread.rs [hunk 2]
```diff
@@ -112,7 +111,11 @@ impl UtilityThreadImpl {
         let mut cum_hashrate_ticker =
             time::interval(Duration::from_secs(CUMULATIVE_HASHRATE_UPDATE_INTERVAL));
 
-        let mut terminal_event_reader = EventStream::new();
+        #[cfg(feature = "crossterm")]
+        let mut terminal_event_reader = crossterm::event::EventStream::new();
+
+        #[cfg(not(feature = "crossterm"))]
+        let mut terminal_event_reader = futures::stream::pending::<()>();
```
[the remark this answers is not in the document]
Comment (mikevoronov) [reply ccp/src/utility_thread/thread.rs:118]: It doesn't connect with the original terminal_event_reader, and doesn't connect with terminal events.

**Pull request merged**